### ISSUE TITLE
Cannot send email with binary attachment on python3

### ISSUE THREAD

**Issue by willstott101** [django-helpdesk/django-helpdesk#502]
Noticed when creating an issue via email and attaching an image (I think).

Simple fix it seems:

```
diff --git a/helpdesk/lib.py b/helpdesk/lib.py
index 4f0ded3..d96a427 100644
--- a/helpdesk/lib.py
+++ b/helpdesk/lib.py
@@ -117,7 +117,7 @@ def send_templated_mail(template_name,
 
     if files:
         for filename, filefield in files:
-            msg.attach(filename, open(filefield.path).read())
+            msg.attach(filename, open(filefield.path, 'rb').read())
 
     return msg.send(fail_silently)
```

Up against it so haven't made a PR yet (cause I assumed you'd want test cases).

Needs tests and a decision on what files / filetypes should be supported and passed through. Or if that's the job of this project or the destination email client/service.

Edit:
A bit more context: `open` in python3 tries decoding the file as unicode text and fails loudly if that fails. This change forces python to read the file as 'raw bytes'. It sounds as though (from the docs) that files with an extension that results in a mimetype starting with `text/` will still be decoded as unicode text by Django, and fallback if that fails. So unicode messages shouldn't actually be affected by this. However I believe it's worth setting up some comprehensive tests for this.

**Comment by gwasser** [comment]
I think we're still working to expand test coverage. Some more tests with example emails that include various attachments is probably necessary to see if this fixes it without regressions, but offhand it makes sense to me.

**Comment by pprolancer** [comment]
Hi @willstott101 and @gwasser 
This fixing will not work for text files in test-cases.
so better fixing is to replace:

msg.attach(filename, open(filefield.path).read())

with

msg.attach_file(filefield.path)

Thanks

**Comment by willstott101** [comment]
I don't see why it wouldn't work with unicode files other than for guessing the incorrect mimetype.

However that file path has no guarantees that it will have the correct extension anyway, right? So the mimetype will still be wrong.

And we need to retain the passed in filename, which we lose with `attach_file` (and incidentally is what `attach` uses to guess the mimetype).

Once considering all that... all the attachments should be received by the server **with** mimetypes shouldn't they? Either via forms or email attachments. Maybe they ought to be passed all the way through to here and included explicitly. There are plenty of mimetypes that Django wont guess right, and passing through whatever we received would be a better bet? Unless that's a security concern? As email clients will probably be set to trust everything coming from helpdesk... maybe that's for another github issue. Worth thinking about though, should we provide an option to exclude attachments from emails and provide links instead (for emails to staff / selected accounts)?

**Comment by gwasser** [comment]
Oh oh oh, it just occurred to me from your discussion above that you were reporting *sent* emails. Oops. I thought you meant the `get_email` script had trouble parsing incoming email, because I know that was a problem in the past.

Probably ideally, everyone would just receive a notification and could log into helpdesk to see the attachment, to prevent potentially malicious things being sent around via email. However, our public portal needs some work before that can happen.

**Comment by reduxionist** [comment]
@pprolancer is correct that the fix in question fails the case as described, and caught, by our tests. unfortunately, the fix he recommends, where is the technically correct one I am I've been waiting to use where wer is that all me the correct one is the correct one once we can drop support for Django 181.8 it's a issue I have document it many times in about jangles not using not using the standard life  libraries email processing module after they moved to python three now that 1.11 has been released we are technically within our church cleaned

**Comment by reduxionist** [comment]
Damn ios dictation!

**Comment by reduxionist** [comment]
In brief while on my mobile: we want to use the case @pprolancer mentions but it fails for 1.8 on py2 iirc, which we can drop support for now that dj 1.11 has been released. Sorry for the nonsense mispost, last time i try to use dictation on my phone! 😊

**Comment by gwasser** [comment]
Recently merged #532 merges in your proposed changes with some updates. I'll close the ticket but reopen or file a new one if there's still trouble. Thanks!

**Comment by willstott101** [comment]
LGTM, thanks for the update.
